contracts/contracts/vault/OETHVaultCore.sol

```diff
@@ -145,4 +157,230 @@ contract OETHVaultCore is VaultCore {
+
+        emit WithdrawalClaimed(msg.sender, requestId, request.amount);
+
+        return request.amount;
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /// @notice Collects harvested rewards from the Dripper as WETH then
+    /// adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// @dev is called from the Native Staking strategy when validator withdrawals are processed.
+    /// It also called before any WETH is allocated to a strategy.
+    function addWithdrawalQueueLiquidity() external {
+        // Stream any harvested rewards (WETH) that are available to the Vault
+        IDripper(dripper).collect();
+
+        _addWithdrawalQueueLiquidity();
+    }
+
+    /// @dev Adds WETH to the withdrawal queue if there is a funding shortfall.
+    /// This assumes 1 WETH equal 1 OETH.
+    function _addWithdrawalQueueLiquidity()
+        internal
+        returns (uint256 addedClaimable)
+    {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+
+        // No need to get WETH balance if all the withdrawal requests are claimable
+        if (queueShortfall > 0) {
+            uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+
+            // Of the claimable withdrawal requests, how much is unclaimed?
+            uint256 unclaimed = queue.claimable - queue.claimed;
+            if (wethBalance > unclaimed) {
+                uint256 unallocatedWeth = wethBalance - unclaimed;
+
+                // the new claimable amount is the smaller of the queue shortfall or unallocated weth
+                addedClaimable = queueShortfall < unallocatedWeth
+                    ? queueShortfall
+                    : unallocatedWeth;
+                uint256 newClaimable = queue.claimable + addedClaimable;
+
+                // Store the new claimable amount back to storage
+                withdrawalQueueMetadata.claimable = uint128(newClaimable);
+
+                // emit a WithdrawalClaimable event
+                emit WithdrawalClaimable(newClaimable, addedClaimable);
+            }
+        }
+    }
+
+    /***************************************
+                View Functions
+    ****************************************/
+
+    function _wethAvailable() internal view returns (uint256 wethAvailable) {
+        WithdrawalQueueMetadata memory queue = withdrawalQueueMetadata;
+
+        // Check if the claimable WETH is less than the queued amount
+        uint256 queueShortfall = queue.queued - queue.claimable;
+        uint256 wethBalance = IERC20(weth).balanceOf(address(this));
+        // Of the claimable withdrawal requests, how much is unclaimed?
+        uint256 unclaimed = queue.claimable - queue.claimed;
+
+        if (wethBalance > queueShortfall + unclaimed) {
+            wethAvailable = wethBalance - queueShortfall - unclaimed;
```

Comment:
I think that this can be simplified in: `wethAvailable = wethBalance  + queue.claimed - queue.queued`, this can avoid 2 unnecessary operations.